File: TPs/TP1/TP01Q10 - Arquivo em C/arquivo.c
#include <stdio.h>

void read(int n){ //Função de leitura do arquivo
    double num = 0;
    int numInt, aux;
    FILE *fp = fopen("arquivo.bin", "rb"); //abertura do arquivo para leitura binária
    if(fp == NULL){ //teste para verificar se o arquivo existe
        printf("Arquivo inválido!\n");
    }else{
        for(int i = 0; i < n; i++){
            fseek(fp, (n-1-i) * 8, SEEK_SET); //procura números de 8 bits
            fread(&num, sizeof(double), 1, fp); // salva o número na variável num
            aux = num;
            if(num - aux != 0) // caso o número for real printa o num
                printf("%g\n", num);
            else //caso xontrário printa aux
                printf("%d\n", aux);
        }
        fclose(fp); //fecha o arquivo
    }
}

void open(int n){ //Função de abertura do arquivo
    double num = 0;
    FILE *fp = fopen("arquivo.bin", "wb"); //abertura do arquivo para escrita binária
    if(fp == NULL){ //teste para verificar se o arquivo existe
        printf("Arquivo inválido!\n");
    }else{
        for(int i = 0; i < n; i++){
            scanf("%lf", &num); //leitura de um número da entrada padrão
            fwrite(&num, sizeof(double), 1, fp); //escreve o que foi lido da entrada padrão no arquivo
        }
        fclose(fp); //fecha o arquivo
    }
}

int main() {
    int num;
    scanf("%d", &num); //leitura do número
    open(num); //chamada da função de abertura do arquivo
    read(num); //chamada da função de leitura do arquivo
}

File: TPs/TP1/TP01Q09 - Arquivo em Java/Arquivo.java
import java.io.File;
import java.io.RandomAccessFile;

public class Arquivo{
    public static int read(int n){ //Leitura do arquivo
        File f = new File("arquivo.bin"); //abertura do arquivo para leitura
        try{ //tratamento de exceções
            if(f == null){ //verifica se é possível abrir o arquivo
                MyIO.println("Não foi possível abrir o arquivo!!!");
                return -1;
            }
            RandomAccessFile file = new RandomAccessFile(f, "r"); //abertura RAF
            for(int i = 0; i < n; i++){
                file.seek((n - 1 - i) * 8); //busca números através do método seek
                double num = file.readDouble(); //lê os números do arquivo e salva na variável num
                if (num % 1 != 0)
                    MyIO.println(num); //printa a variável num na saída padrão
                else
                    MyIO.println((int) num); //converte a variável num para inteiro e printa na saída padrão
            }
            file.close(); //fecha o arquivo
        } catch(Exception e){ //pega as exceções e salva no endereço de memória de e

        }
        return 0;
    }

    public static int open(int n){ //abre arquivo para escrita
        double num = 0;
        File f = new File("arquivo.bin");
            try{ //tratamento de exceções
                if(f == null){ //verifica se é possível abrir o arquivo
                    MyIO.println("Não foi possível abrir o arquivo!!!");
                    return -1;
                }
                RandomAccessFile file = new RandomAccessFile(f, "rw"); //abertura RAF
                for(int i = 0; i < n; i++){
                    num = MyIO.readDouble(); //salva o número lido no arquivo na variável num
                    file.writeDouble(num); //escreve a variável num no arquivo
                }
                file.close(); //fecha o arquivo
        }catch(Exception e){ //pega as exceções e salva no endereço de memória de e

        }
        return 0;
    }

    public static void main(String[] args){ //função main
        int num = MyIO.readInt();
        open(num); //chamada da função de abertura de arquivo
        read(num); //chamada da função de leitura de arquivo
    }
}